### ISSUE TITLE
Problem with analyseBBlocks and intrinsics / function call disambiguation and flows graph

### ISSUE THREAD

**Issue by dorchard** [camfort/fortran-src#51]
Consider the following example from `camfort`:
```
program bug
  implicit none
  integer :: i
  real, dimension(2) :: a, b
  real :: x
  do i = 1, 2
    x = abs(a(i))
    b(i) = x
  end do
end program
```
We should infer a specification of `!= stencil readOnce, pointed(dim=1) :: a` here but we don't because there are no subscripts flowing to `b(i)`. I've narrowed this down to a problem with the basic blocks / flows graph I think. Basically, the transformations carried out by `processFunctions` of `BBlocks` don't seem to be interacting well with the flows graph here. Perhaps the newly created statements aren't getting picked up in the flows graph?

**Comment by dorchard** [comment]
Note that you get the correct behaviour here if you remove the intermediate assignment, i.e. `b(i) = abs(a(i))`

**Comment by mrd** [comment]
sorry, closed by 895ada70d3afa031091224a44e978e306a0583ff
